src/main/java/bio/terra/cli/command/resources/update/GcsBucket.java

```diff
@@ -0,0 +1,69 @@
+package bio.terra.cli.command.resources.update;
+
+import bio.terra.cli.businessobject.Context;
+import bio.terra.cli.businessobject.Resource;
+import bio.terra.cli.command.shared.BaseCommand;
+import bio.terra.cli.command.shared.options.Format;
+import bio.terra.cli.command.shared.options.GcsBucketStorageClass;
+import bio.terra.cli.command.shared.options.ResourceUpdate;
+import bio.terra.cli.command.shared.options.WorkspaceOverride;
+import bio.terra.cli.exception.UserActionableException;
+import bio.terra.cli.serialization.userfacing.inputs.UpdateGcsBucketParams;
+import bio.terra.cli.serialization.userfacing.resources.UFGcsBucket;
+import bio.terra.workspace.model.StewardshipType;
+import picocli.CommandLine;
+
+/** This class corresponds to the fourth-level "terra resources update gcs-bucket" command. */
+@CommandLine.Command(
+    name = "gcs-bucket",
+    description = "Update a GCS bucket.",
+    showDefaultValues = true)
+public class GcsBucket extends BaseCommand {
+  @CommandLine.Mixin ResourceUpdate resourceUpdateOptions;
+  @CommandLine.Mixin GcsBucketStorageClass storageClassOption;
+  @CommandLine.Mixin bio.terra.cli.command.shared.options.GcsBucketLifecycle lifecycleOptions;
+
+  @CommandLine.Mixin WorkspaceOverride workspaceOption;
+  @CommandLine.Mixin Format formatOption;
+
+  /** Update a Big Query dataset in the workspace. */
+  @Override
+  protected void execute() {
+    workspaceOption.overrideIfSpecified();
+
+    // all update parameters are optional, but make sure at least one is specified
+    if (!resourceUpdateOptions.isDefined()
+        && !storageClassOption.isDefined()
+        && !lifecycleOptions.isDefined()) {
```

Comment:
can we do that with an ArgGroup? or does that not work with mixins, I forget.

Reply:
You can have an ArgGroup that requires at least one option, though the usage help looks pretty weird. e.g.
`([-a=<a>] [-b=<b>] [-c=<c>]) [[-a=<a>] [-b=<b>] [-c=<c>]] [[-a=<a>] [-b=<b>] [-c=<c>]]`

But it doesn't work with mixins (compile time exception `@Mixins are not supported on @ArgGroups`).